### ISSUE TITLE
[hmac,dv] Compare intermediate digests when verifying HMAC save & restore

### ISSUE THREAD

**Issue by gdessouky** [lowRISC/opentitan#23240]
### Description

Implement checks to compare intermediate digests with expected intermediate digests from C models when verifying the save & restore HMAC feature - see https://github.com/lowRISC/opentitan/pull/23116#issuecomment-2112015540.

**Comment by martin-velay** [comment]
### Notes for the person who will make the modifications

I have printed out the the intermediate digest from the C model for a debug purpose. I am just adding here how, to be able to progress quickly.

After this [line](https://cs.opensource.google/opentitan/opentitan/+/master:hw/ip/hmac/dv/cryptoc_dpi/hmac_wrap.c;l=29) from hmac_wrap.c file, I added this:
```
  vpi_printf("Debug - Intermediate HASH from C model:\n");
  for (int i=0; i<8; i++) {
    vpi_printf("%08x, ", ctx.hash.state[i]);
  }
  vpi_printf("\n");
```
This was only to debug SHA2-256, do similar stuff for other digest sizes and see how to make it visible to the scoreboard for auto checking.

Another point, from the scoreboard a condition has been added to detect when the digest is updated because of a stop command, and in that case the digest is not compared as the RTL is sharing the intermediate value. When this value is available from the SCB, remember to make this check possible. Modification should occur where there is these comments:
```
          // Predict intermediate digest values when S&R has been triggered
          // TODO later, modify the C model to be able to provide intermediate digest (PR #23116)
          // if (hmac_stopped) predict_digest(msg_q);
          // TODO remove lines, when C model will support intermediate hash. For the moment don't
          // compare digest. Only the final digest will be compared in case of S&R, this is enough
          // for the moment
          // Proceed with read digest and compare values with expected
          if (!hmac_stopped) begin
```

**Comment by andreaskurth** [comment]
I think this can be moved to V3 and M7; WDYT @martin-velay?

**Comment by martin-velay** [comment]
@andreaskurth: yes we can move it as here we are not going to find bug as this will only help the DV in case a debug is needed at this level

**Comment by martin-velay** [comment]
This won't bring more confidence, it's more for debugging in case of an issue is raised. Could be moved into M7 backlog

**Comment by martin-velay** [comment]
Not required for D3/V3 as it won't increase our confidence on the IP or change anything on the coverage. But should be done later as it can be used in case of a bug is found.
